erlang/pop_protocol/pop_protocol.erl

```diff
@@ -2,14 +2,40 @@
 
 -export([add_one_block/3, get_genesis_tree_data/1]).
 
--import(blocktree, [add_new_block/2, get_block_by_id/2]).
--import(my_crypto, [hash/1, sign/2, verify/3]).
--import(my_serializer, [serialize_object/1]).
+%% -import(blocktree, [add_new_block/2, get_block_by_id/2]).
+%% -import(my_crypto, [hash/1, sign/2, verify/3]).
+%% -import(my_serializer, [serialize_object/1]).
 
 -include_lib("stdlib/include/assert.hrl").
 
 -include("../potato_records.hrl").
 
+map_key_match_assert(Map, KeyList) ->
+    K1 = lists:sort(maps:keys(Map)),
+    K2 = lists:sort(KeyList),
+    ?assertEqual(K1, K2, {"key mismatch", K1, K2}),
+    ok.
+
+transaction_map_structure_assert(T) when is_map(T) ->
+    map_key_match_assert(T, [game_data, nonce, player_id, consensus_data]),
+    
+    CD = maps:get(consensus_data, T),
+    map_key_match_assert(CD, [signature, chain_id]),
+    
+    ok.
+
+check_block_map_structure(B) when is_map(B) ->
```

Comment:
Basically records :D

Maybe it should be like check_map_structure(structure, map)
where structure are vars with value [key1,key2,key3,...] and predefined in some constants file.

Reply:
like in potato_records.hrl? O: O: O:

Reply:
(unless that was the joke you were making)